### PR TITLE
Support for gzip, zip and deflate headers for LOAD CSV

### PR DESCRIPTION
**Pull request by pontusmelke**

### CODE REVIEW

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/spi/CSVResources.scala [hunk 1]
```diff
@@ -97,6 +103,12 @@ class CSVResources(cleaner: TaskCloser) extends ExternalResource {
       con.setConnectTimeout(connectionTimeout)
       con.setReadTimeout(readTimeout)
       con.getInputStream
+      val stream = con.getInputStream
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/spi/CSVResources.scala]: I don't know how to write test for this part?

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/spi/CSVResources.scala [hunk 2]
```diff
@@ -97,6 +103,12 @@ class CSVResources(cleaner: TaskCloser) extends ExternalResource {
       con.setConnectTimeout(connectionTimeout)
       con.setReadTimeout(readTimeout)
       con.getInputStream
```
Comment (craigtaverner) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/spi/CSVResources.scala]: I guess this line should go?

**Comment by nawroth** [comment]
@pontusmelke 
This looks like yet one more thing to mention around here: https://github.com/neo4j/neo4j/blame/2.3/manual/cypher/cypher-docs/src/docs/dev/ql/load-csv/index.asciidoc#L9

**Comment by Mats-SX** [comment]
Rebased.

**Pull request merged**